### PR TITLE
Add support for Mount Targon

### PR DESCRIPTION
**Pull request by dnorth**
This ends up being very similar to #8 but different in a few ways.

1. This is still using version 2. According to [Riot documentation](https://github.com/RiotGames/LoRDeckCodes#process) version 3 is not yet supported in the client and so that PR will break codes if pushed as-is. 

2. I changed the way faction shortCodes and ids are organized. Rather than using an array it is now using an object. I think this has a few benefits:
    1. When a new region is added, it is much easier to make sure the id is what it is expected to be. Previously, because it was an array the array needed to be filled with null values in order to make sure the region was in the correct location. Imagine in the future a region has id 104 for whatever reason. The array would then need to be 105 items long and make sure that each region is in its correct spot. 
    2. It's much easier to add new regions. When it is an object you can just add it to the end of the object as a key/value pair and not worry about where to put that new region. So whenever a new region is released it is trivial to add the region to the list.

If you prefer the array way then I can put that back, but I figured i'd propose this new way which I think will be easier to digest for future updates.

### CODE REVIEW

File: src/Faction.js
Comment (dnorth) [inline src/Faction.js:11]: I updated the message here so that the user is signaled to update this package. Wording can be modified if wanted.

File: src/Faction.js
Comment (dnorth) [inline src/Faction.js:21]: Added an error here as well. This should also be failing for the same reason as above because if not it will return a completely bogus faction that doesn't make sense.

**Comment by pedrofracassi** [comment]
Uh, I wonder why none of our tests ran here. Might have to do with permissions.

**Comment by pedrofracassi** [comment]
> According to Riot documentation version 3 is not yet supported in the client and so that PR will break codes if pushed as-is.

That's very weird, Tuxedo removed the warning from their README and it seems that we'll use v2 with MT support for the rest of the patch (https://github.com/RiotGames/LoRDeckCodes/commit/84217630c1021b685a0446c2eb718c5cb8fe3c73)

> If you prefer the array way then I can put that back, but I figured i'd propose this new way which I think will be easier to digest for future updates.

No need to. I wonder why we didn't use an object in first place 😂

---

Changes look good to me, I'll get our CI to work here so we can run tests and then wait for @davipatury's review so we can merge (he should wake up in a few hours).

**Review by pedrofracassi — APPROVED** [review]
LGTM!

**Comment by pedrofracassi** [comment]
I've modified the GitHub Actions config file so that we can run tests on Pull Requests. Please pull the latest changes from our master branch :)

**Comment by dnorth** [comment]
Alright, pulled in your changes and it looks like tests are running now :)

**Comment by dnorth** [comment]
All checks passed, just pending review by @davipatury

**Comment by dnorth** [comment]
Last question @pedrofracassi I think we also need a version update and then a package upgrade when this is merged in. Do you want me to add a version update commit, or do you want to do that after-the-fact? (And if you want me to do it, would you consider this a minor or major update?)

**Comment by pedrofracassi** [comment]
> Do you want me to add a version update commit, or do you want to do that after-the-fact? (And if you want me to do it, would you consider this a minor or major update?)

Our GitHub Actions config automatically releases versions to NPM as soon as we change the version in package.json, so I guess I'll do it myself after we merge this.

**Review by davipatury — APPROVED** [review]
LGTM

**Pull request merged**
